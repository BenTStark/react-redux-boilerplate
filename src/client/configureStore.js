import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import { routerMiddleware } from "connected-react-router";
import reducer from "./reducer";

export default function configureStore(history, initialState = {}) {
  const reduxRouterMiddleware = routerMiddleware(history);
  const middleware = [thunk, reduxRouterMiddleware];

  const enhancer = compose(
    // Middleware you want to use in development:
    applyMiddleware(thunk, reduxRouterMiddleware),
    // Required! Enable Redux DevTools with the monitors you chose
    window.devToolsExtension ? window.devToolsExtension() : f => f
  );

  // Sync dispatched route actions to the history

  const store = createStore(reducer(history), initialState, enhancer);

  // if (module.hot) {
  //   module.hot.accept('../reducers', () => {
  //     store.replaceReducer(require('./../reducers').default);
  //   });
  // }
  return store;
}
