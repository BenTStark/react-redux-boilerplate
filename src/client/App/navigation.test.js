// als Beispiel wie man mounten kÃ¶nnte:
// https://github.com/MacKentoch/react-redux-bootstrap-webpack-starter
import React from "react";
import NavigationComponent from "./navigation.component";
import NavigationContainer from "./navigation.container";
import configureStore from "redux-mock-store";
import toJson from "enzyme-to-json";
import MockAdapter from "axios-mock-adapter";
import { shallow } from "enzyme";

describe(">>> NavigationComponent - Snapshot", () => {
  it("+++capturing Snapshot of NavigationComponent", () => {
    const auth = {};
    auth.loginSuccess = true;
    auth.profile = { nickname: "John Doe" };
    // Shallow because lot of strange errors with mount
    const renderedValue = shallow(<NavigationComponent auth={auth} />);
    expect(toJson(renderedValue)).toMatchSnapshot();
  });
});

// Component Test
describe(">>> AppComponent --- Shallow Render React Components", () => {
  const wrapper = shallow(<NavigationComponent auth={{}} />);

  it("+++ render the COMPONENT", () => {
    expect(wrapper.length).toEqual(1);
  });
});


// Container Test
describe(">>> NavigationContainer --- REACT-REDUX (Shallow + passing the {store} directly)", () => {
  const initialState = {
    authReducer: {
      profile: null,
      accessToken: null,
      loginRequest: false,
      loginSuccess: false,
      loginError: false
    }
  };
  const mockStore = configureStore();
  let store, container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<NavigationContainer store={store} />);
  });

  it("+++ check Prop matches with initialState", () => {
    expect(container.prop("auth")).toEqual(initialState.authReducer);
  });
});
