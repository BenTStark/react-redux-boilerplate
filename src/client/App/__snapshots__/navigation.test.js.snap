// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> NavigationComponent - Snapshot +++capturing Snapshot of NavigationComponent 1`] = `
<div>
  <Uncontrolled(Navbar)
    bsStyle="default"
    fixedTop={true}
  >
    <NavbarHeader>
      <NavbarBrand>
        <LinkContainer
          activeClassName="active"
          className="title"
          exact={false}
          replace={false}
          strict={false}
          to="/"
        >
          <div>
            <span>
              React-Redux-Boilerplate
            </span>
          </div>
        </LinkContainer>
      </NavbarBrand>
      <NavbarToggle />
    </NavbarHeader>
    <NavbarCollapse>
      <Nav
        bsClass="nav"
        justified={false}
        navbar={true}
        pullLeft={false}
        pullRight={false}
        stacked={false}
      >
        <NavItem
          active={false}
          disabled={false}
          onClick={[Function]}
        >
          Logout
        </NavItem>
        <LinkContainer
          activeClassName="active"
          exact={false}
          replace={false}
          strict={false}
          to="/otherRoute"
        >
          <NavItem
            active={false}
            disabled={false}
          >
            Other Route
          </NavItem>
        </LinkContainer>
        <LinkContainer
          activeClassName="active"
          exact={false}
          replace={false}
          strict={false}
          to="/blog"
        >
          <NavItem
            active={false}
            disabled={false}
          >
            Blog
          </NavItem>
        </LinkContainer>
        <LinkContainer
          activeClassName="active"
          exact={false}
          replace={false}
          strict={false}
          to="/contact"
        >
          <NavItem
            active={false}
            disabled={false}
          >
            Contact
          </NavItem>
        </LinkContainer>
      </Nav>
      <p
        className="navbar-text"
      >
        Logged in as 
        <strong>
          John Doe
        </strong>
        .
      </p>
      <Nav
        bsClass="nav"
        justified={false}
        navbar={true}
        pullLeft={false}
        pullRight={true}
        stacked={false}
      >
        <NavItem
          active={false}
          disabled={false}
          href="https://github.com/BenTStark/react-redux-boilerplate"
          target="_blank"
          title="View on Github"
        >
          <i
            className="fa fa-github"
          />
        </NavItem>
      </Nav>
    </NavbarCollapse>
  </Uncontrolled(Navbar)>
</div>
`;
